### PR TITLE
Next version

### LINKED ISSUE #26: `sandbox` should not coerce to `Rel`
Right now, `sandbox` is merely `flip relativeTo`, which turns any path into `Rel`. Shouldn't it return whatever the original path was, merely flipping `Unsandboxed` to `Sandboxed`?

### LINKED ISSUE #28: Use custom kinds for the phantom types
To ensure the `Abs` and `Sandboxed` aren't mixed up in a type signature, for example.

`DataKinds` would be even better, but we don't have those, so :wink:

### LINKED ISSUE #30: Instead of having `FileName` and `DirName` introduce `Name` that can be `Name Dir` / `Name File`
This means when we `peel` we don't have to account for a case that is known to be impossible, as `peel`ing a `DirPath` would give you a `Name Dir` with:

``` purescript
peel
  :: forall a b s
   . Path a b s
-> Maybe (Tuple (Path a Dir s) (Name b))
```

### PR DESCRIPTION
**Pull request by safareli**
fixes #26
fixes #28
fixes #30

### CODE REVIEW

**Comment by safareli** [comment]
The `view*` functions I added are somehow related to:

> Introduce a powerful fold method and use that to implement all the "unsafe" (type parameter-changing) operations. [from](https://github.com/slamdata/purescript-pathy/issues/19#issuecomment-234789621)

**Comment by safareli** [comment]
There are some notes in https://github.com/safareli/purescript-pathy/pull/2

File: src/Pathy/Name.purs [hunk 1]
```diff
@@ -24,32 +24,72 @@ derive newtype instance ordName :: Ord (Name a)
...
   show (Name name) = "(Name " <> show name <> ")"
 
--- | Retrieves the extension of a name.
+-- | Splits `Name` in name and extension part.
+-- |
+-- | ```purescript
+-- | splitName (Name ".foo")    == { name: ".foo", extension: Nothing }
+-- | splitName (Name "foo.")    == { name: "foo.", extension: Nothing }
+-- | splitName (Name "foo")     == { name: "foo",  extension: Nothing }
+-- | splitName (Name ".")       == { name: ".",    extension: Nothing }
```
Comment (safareli) [inline src/Pathy/Name.purs:33]: We can remove this example as it shuoldn't be actually used right? We can constrain this in the `IsName` class too, @garyb what you think?

**Comment by garyb** [comment]
I think that's it! Unless you there's anything else you can think of?

**Review by garyb — APPROVED** [review]
:tada:

**Pull request merged**